docs/admin/extensible-admission-controllers.md

```diff
@@ -20,11 +21,107 @@ the following:
 * They need to be compiled into kube-apiserver.
 * They are only configurable when the apiserver starts up.
 
-1.7 introduces two alpha features, *Initializers* and *External Admission
-Webhooks*, that address these limitations. These features allow admission
-controllers to be developed out-of-tree and configured at runtime.
+Two features, *Admission Webhooks* (beta in 1.9) and *Initializers*, 
+address these limitations. They allow admission controllers to be developed 
+out-of-tree and configured at runtime.
 
-This page describes how to use Initializers and External Admission Webhooks.
+This page describes how to use Admission Webhooks and Initializers.
+
+## Admission Webhooks
+
+### What are admission webhooks?
+
+Admission webhooks are HTTP callbacks that receive admission requests and do
+something with them. You can define two types of admission webhooks,
+[ValidatingAdmissionWebhooks](/docs/admin/admission-controllers.md#validatingadmissionwebhook-alpha-in-18-beta-in-19)
+and [MutatingAdmissionWebhooks](/docs/admin/admission-controllers.md#mutatingadmissionwebhook-beta-in-19).
+With `ValidatingAdmissionWebhooks`, you may reject requests to enforce custom 
+admission policies. With `MutatingAdmissionWebhooks`, you may change requests 
+to enforce custom defaults.
+
+### Experimenting with admission webhooks
+
+Admission webhooks are essentially part of the cluster control-plane. You should
+write and deploy them with great caution. Please read the [user
+guides](https://github.com/kubernetes/website/pull/6836/files)(WIP) for
+instructions if you intend to write/deploy production-grade admission webhooks.
+In the following, we describe how to quickly experiment with admission webhooks.
+
+### Prerequisites
+
+* Ensure you are on a Kubernetes cluster that is at least as new as v1.9.
+
+* Ensure that MutatingAdmissionWebhook and ValidatingAdmissionWebhook admission
+controllers are enabled. [Here](/docs/admin/admission-controllers.md#is-there-a-recommended-set-of-admission-controllers-to-use)
+is a recommended set of admission controllers to enable in general.
+
+* Ensure that the admissionregistration.k8s.io/v1beta1 API is enabled.
+
+### Write a admission webhook server
+
+Please refer to the implementation of the [admission webhook
+server](https://github.com/kubernetes/kubernetes/blob/v1.10.0-beta.1/test/images/webhook/main.go)
+that is validated in Kubernetes e2e test. The webhook handles the
+`admissionReview` requests sent by the apiservers, and sends back its decision
+wrapped in `admissionResponse`.
+
+Note that by v1.10 you can only dynamically configure the
+[CABundle](https://github.com/kubernetes/kubernetes/blob/v1.10.0-beta.1/staging/src/k8s.io/api/admissionregistration/v1beta1/types.go#L263)
+that is used to validate the cert present by the webhooks to the apiservers, but
+not the CA bundles for the other direction, so the example webhook does not
```

Comment:
the client (the apiserver) doesn't send a CA bundle to the webhook, it sends client credentials (cert or token). need a section here or in the admission config doc describing how you provide credentials to apiservers for them to use to contact admission webhooks:

```yaml
kind: AdmissionConfiguration
apiVersion: apiserver.k8s.io/v1alpha1
plugins:
- name: MutatingAdmissionWebhook
  configuration:
    apiVersion: apiserver.config.k8s.io/v1alpha1
    kind: WebhookAdmission
    kubeConfigFile: /path/to/kubeconfig
- name: ValidatingAdmissionWebhook
  configuration:
    apiVersion: apiserver.config.k8s.io/v1alpha1
    kind: WebhookAdmission
    kubeConfigFile: /path/to/kubeconfig
...
```

kubeconfig
```yaml
apiVersion: v1
users:
# use a client certificate by default
- name: *
  user:
    client-certificate-data: <data>
    client-key-data: <data>
# use some other credentials for particular domains
- name: webhook1.ns1.svc
  user:
    token: 123123123123
```

Reply:
needs to be be ```- name: '*'``` in the kubeconfig

Reply:
Ack.

The `AdmissionConfiguration` isn't dynamically configurable, right? It's a static file that user supplier to the apiserver when it's started.

Reply:
I have tested these out locally and updated the doc locally. I'll find time this week to finish addressing the comments.